src/write_resume_data.cpp

```diff
@@ -98,11 +98,10 @@ namespace libtorrent {
 
 		if (!atp.merkle_trees.empty())
 		{
-			TORRENT_ASSERT(atp.info_hash.has_v2());
```

Comment:
this seems like a generally reasonable assert, but I think it's nice if `read_resume_data()` and `write_resume_data()` don't enforce any specific invariant of the resume data. It could be asserted deeper in the torrent or session, when loading the resume data.